### ISSUE TITLE
PRM

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/trl#2631]

**Comment by qgallouedec** [comment]
```python
from datasets import load_dataset
from transformers import AutoModelForTokenClassification, AutoTokenizer

from trl import PRMConfig, PRMTrainer

model_id = "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5"
model = AutoModelForTokenClassification.from_pretrained(model_id)
tokenizer = AutoTokenizer.from_pretrained(model_id)

dummy_dataset = load_dataset("trl-internal-testing/zen", "standard_stepwise_supervision", split="train")


batch_size = 4
gradient_accumulation_steps = 2
output_dir = f"PRM-bsz{batch_size}-grad_acc{gradient_accumulation_steps}-fixed"

training_args = PRMConfig(
    output_dir=output_dir,
    per_device_train_batch_size=batch_size,
    gradient_accumulation_steps=gradient_accumulation_steps,
    logging_steps=2,
)
trainer = PRMTrainer(model=model, args=training_args, processing_class=tokenizer, train_dataset=dummy_dataset)

trainer.train()
```

We don't get the issue as we don't compute the loss by our own.

![Image](https://github.com/user-attachments/assets/ea60a022-9b82-4e5d-8cea-8017c7dba10a)
